deps/parselibs.jl

```diff
@@ -0,0 +1,46 @@
+function execute(cmd::Cmd)
+    out = Pipe()
+    err = Pipe()
+  
+    process = run(pipeline(ignorestatus(cmd), stdout = out, stderr = err))
+    close(out.in)
+    close(err.in)
+  
+    (stdout = String(read(out)), 
+      stderr = String(read(err)),  
+      code = process.exitcode)
+end
+
+libparsepath = abspath(joinpath(@__DIR__, "..", "local", "bin", "libparse"))
+
+library_dir = ""
+
+if haskey(ENV, "SINGULAR_LIBRARY_DIR")
+    library_dir = ENV["SINGULAR_LIBRARY_DIR"]
+else
+    library_dir = abspath(joinpath(@__DIR__, "..", "local", "share", "singular", "LIB"))
+end
+
+filenames = filter(x -> endswith(x, ".lib"), readdir(library_dir))
+
+output_filename = abspath(joinpath(@__DIR__, "..", "src", "libraryfuncdictionary.jl"))
+
+open(output_filename, "w") do outputfile
+    println(outputfile, "libraryfunctiondictionary = Dict(")
+    for i in filenames
+        full_path = joinpath(library_dir, i)
+        libs = execute(`$libparsepath $full_path`)
+        if libs.stderr != ""
+            error("from libparse: $(libs.stderr)")
+        end
+        libs_splitted = split(libs.stdout,"\n")[4:end-1]
```

Comment:
I think a comment here would be very helpful; something like this:
```julia
# The output of libparse looks as follows: each line has the form ...
#  .... give generic description of the form ...
#  For a concrete example, it might be this:
#  ... concrete example ...
#  From this, we need to extra ...
```